scripts/update_tags_per_languages.py

```diff
@@ -0,0 +1,365 @@
+    # tag field in API, file name in taxonomies: url parameter
+    # "allergens": "allergen",
+    # "brands": "brand", # all unknown for now (no taxonomy for it)
+    "categories": "category",
+    # "countries": "country",
+    # "labels": "label",
+    # "origins": "origin", 
+}
+
+dev = True
+if dev:
+    env = "net"
+    user = "off:off@" 
+else:
+    env = "org"
+    user = "" 
+
+headers = {
+    'Accept': 'application/json', 
+    'User-Agent': 'UpdateTagsLanguages',
+}
+data = {
+    'user_id': '',
+    'password': '',
+}
+
+logs_file = "update_tags_per_languages_logs"
+if os.path.isfile(logs_file):
+    os.remove(logs_file)
+
+mapping_languages_countries = {
+    "aa": "dj",
+    "ar": "world",  # ar but categories are en:<french name> 
+    "be": "by",
+    "bg": "bg",
+    "br": "fr",
+    "bs": "ba",
+    "ca": "fr",
+    "cs": "cz",
+    "da": "dk",
+    "de": "de",
+    "el": "gr",
+    "en": "world",
+    "xx": "world", # xx but categories are en:<french name>
+    "es": "es",
+    "et": "ee",
+    "fa": "ir",
+    "fi": "fi",
+    "fr": "fr",
+    "hr": "hr",
+    "id": "id",
+    "is": "is",
+    "it": "it",
+    "ja": "jp",
+    "lt": "lt",
+    "ms": "my",
+    "nb": "no",
+    "nl": "nl",
+    "pl": "pl",
+    "pt": "pt",
+    "ro": "ro",
+    "ru": "ru",
+    "sk": "sk",
+    "sl": "si",
+    "sr": "rs",
+    "sv": "se",
+    "th": "th",
+    "zh": "cn",
+}
+
+def get_from_api(get_call_url: str) -> dict:
+    """Send a GET request to the given URL.
+    
+    :param get_call_url: the URL to send the request to
+
+    :return: the API response
+    stop the code if returned status_code is different than 200
+    """
+    get_call_url_res = requests.get(
+        get_call_url,
+        headers=headers,
+    )
+
+    if get_call_url_res.status_code != 200:
+        print(f"ERROR: when calling api. {get_call_url_res.status_code} status code. url: {get_call_url}")
+        sys.exit()
+    
+    return get_call_url_res.json()
+
+
+def unknown_tags_taxonomy_comparison(all_tags: dict, file: str, tag_type: str) -> dict:
+    """Iterate over all referenced tags, filter unknowns tags, and search in taxonomy
+
+    :param all_tags: all_tags as returned by API call
+    :param file: path of the corresponding taxonomy file
+    :param tag_type: the current tag type (category, allergens, etc.). Used to save output file
+
+    :return: possible_wrong_language_tags {"old_tag": "tag_found_in_different_lang"}
+    """
+    possible_new_tags = []
+    possible_wrong_language_tags = {}
+    already_referenced_tags = []
+
+    for tag in all_tags["tags"]:
+        if tag["known"] == 0:
+            # limit number of iterations
+            if dev and len(possible_wrong_language_tags) > 0:
+                break
+
+            tag_name = tag['name']
+            # should retrieve all "en:blablabla, tag_name" or "it:tag_name"
+            tag_regex = re.compile(f'\n([a-z][a-z]:(?:[\w\s\-\']*\,-)*{tag_name})[,|\n]')
```

Comment:
I'm not sure why we include a comma in it ? Why don't we stop on commas ?
Is this to ensure, tag_name appear in a synonym list ?

Reply:
f'\n([a-z][a-z]:(?:[\w\s\-\']*\,-)*{tag_name})[,|\n]'

There are 2 commas in the regex.

if you have:

> en:Hamburger buns, Burger buns, **Rolls for hamburger**, Buns for hamburger

and you are searching for "Rolls for hamburger"
the first comma in the regex would ignore the comma at the end of

> en:Hamburger buns, Burger buns, 

the second comma in the regex would ignore the comma right after the occurence:

> , Buns for hamburger

Reply:
Ok, I see, I'll try to comment, feel free to change:

```suggestion
            # should retrieve all "en:blablabla, tag_name" or "it:tag_name"
            # the prefix is either the language or a comma. 
            # Suffix is either an end of line or a comma
            tag_regex = re.compile(f'\n([a-z][a-z]:(?:[\w\s\-\']*\,-)*{tag_name})[,|\n]')
```

Do you know you can also use [positive lookahead or lookbehing](https://docs.python.org/3/library/re.html#) (`(?=...)` and `(?<=...)`)

Reply:
interesting

Actual regex captures the whole block: `"([a-z][a-z]:(?:[\w\s\-\']*\,-)*({tag_name})"`
It can be "en:blablabla, tag_name" or "it:tag_name", for example.

In the next lines, it checks if it starts with "en:" or another language, and if not it keeps the first element of the list of synonyms (for example, if we looked for "Alga spirulina" and it found "it:Spirulina, Alga spirulina", then it will keep "it:Spirulina")

Eventually, we could put "(?P<some name>[a-z][a-z]:(?:[\w\s\-\']*\,-)*(?P<tag_name>{tag_name})", but maybe it is good as it is already now.